flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/decorators/DecoratorWithPodTemplateTestBase.java

```diff
@@ -215,4 +215,18 @@ void testMainContainerImagePullPolicyOverwritten() {
         assertThat(this.resultPod.getMainContainer().getImagePullPolicy())
                 .isEqualTo(IMAGE_PULL_POLICY);
     }
+
+    @Test
+    void testDNSPolicyWithPodTemplate() {
+        this.resultPod.getPodWithoutMainContainer().getSpec().setDnsPolicy("None");
```

Comment:
I suggest to remove this line and add ` dnsPolicy: None` to the `testing-pod-template.yaml`.